### PR TITLE
Improve package tools

### PR DESCRIPTION
**Pull request by mdouchin**
This PR adds new algorithms and improve existing ones to make it easier to prepare data for field work

WIP

### CODE REVIEW

**Comment by Gustry** [comment]
Note for me :
Damned, there is still a travis running, I will kill it ... (I though I killed the last ones last week). Maybe still gobs

**Comment by Gustry** [comment]
@mdouchin I have merged #31, if you want to rebase on top of master, it's up to you ;-)
Tests are passing.

**Review by Gustry — COMMENTED** [review]
Just a quick review for now. It's big PR ;-)

**Comment by Gustry** [comment]
PR rebased to fix the conflict

**Comment by mdouchin** [comment]
@Gustry I added more commits to this --too-- big PR... I would like to merge it and improve things later. 
NB: I have already rebased from updated master

File: lizsync/processing/algorithms/build_mobile_project.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+        connection_name_central = ls.variable('postgresql:central/name')
+        db_param_a = QgsProcessingParameterString(
+            self.CONNECTION_NAME_CENTRAL,
+            tr('PostgreSQL connection to the central database'),
+            defaultValue=connection_name_central,
+            optional=False
+        )
+        db_param_a.setMetadata({
+            'widget_wrapper': {
+                'class': 'processing.gui.wrappers_postgis.ConnectionWidgetWrapper'
```
[the remark this answers is not in the document]
Comment (mdouchin) [reply lizsync/processing/algorithms/build_mobile_project.py:75]: Fichtre :(

File: docs/concepts/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Presentation
+## Presentation
```
Comment (Gustry) [inline docs/concepts/README.md:1]: According to markdown rules, headings should be increasing one by one.
It's semantic against display ;-)

You might have a mkdocs warning for that. Not sure if I have actived.
But not strong opinion.

**Review by Gustry — COMMENTED** [review]
Ok, so just merge ;-)
Sorry !

**Comment by mdouchin** [comment]
@Gustry I made some changes, but I have a problem with the tests:
https://github.com/3liz/qgis-lizsync-plugin/pull/29/checks?check_run_id=1988674506#step:4:9

This rings a bell, but I cannot remember yet how we solve this in the past, or if it was only a bad configuration involved

NB: Mkdoc is great. I played a bit by adding the android menu and move the French doc to the end ;)

**Comment by mdouchin** [comment]
I have just pushed the commit a1842fe to adapt connection database inputs for QGIS >= 3.14

**Comment by Gustry** [comment]
I think, you can merge. It's broken on the master branch as well !

Is-it working on your local machine ?

It's since my latest changes I think. Creating a ticket.

**Comment by mdouchin** [comment]
it also failed locally on this branch and master. I probably have mistakes to correct after beeing able to run the test suite...

**Comment by Gustry** [comment]
> it also failed locally on this branch and master. I probably have mistakes to correct after beeing able to run the test suite...

Sure, we will have a look.
When do you plan this merge ?

I can have a look to fix the master branch.

**Comment by mdouchin** [comment]
I am ready to merge. I manually tested the algs, but some mistakes can still occur as unit tests are not run.

**Comment by Gustry** [comment]
Fixed on master, can you rebase your branch ?
Sorry for that.

**Pull request merged**